src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.CurrentUserOnly.Windows.cs

```diff
@@ -0,0 +1,151 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32.SafeHandles;
+using System.ComponentModel;
+using System.DirectoryServices.AccountManagement;
+using System.Runtime.InteropServices;
+using System.Security.Principal;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.Pipes.Tests
+{
+    // Class to be used as xUnit fixture to avoid creating the user, an relatively slow operation (couple of seconds), multiple times.
+    public class TestAccountImpersonator : IDisposable
+    {
+        private const string TestAccountName = "CorFxTst0uZa"; // Extra random suffix to avoid matching any other account by accident, but const to avoid leaking it.
```

Comment:
Does this mean that when I run tests locally, I end up creating an account with a known username and password that someone could use to access my machine?

Reply:
The name is fixed but the password is generated on every run using `Guid.NewGuid()` (see line 24). The account also expires after 5 minutes, which can be reduced, granted that expiration doesn't help much if a machine is constantly running the tests. I also looked into disabling remote login with this account but I didn't find a straightforward way of doing that, but I will keep looking at it or any other ideas to harder the scenario.

The alternative to creating this account that I considered was using Windows services, I can consider that path again if the current one seems too risky.

Reply:
> the password is generated on every run using Guid.NewGuid()

Ah, I missed that.  Ok, thanks.

> the password is generated on every run using Guid.NewGuid()

Great. Thanks.

Reply:
Guid.NewGuid() is not cryptographically secure. A cheap alternative is call Path.GetRandomFileName()...Do it twice and concatenate to get a reasonable length.

Reply:
For quite some time `UuidCreate` is using crypto but then I don't remember exactly when that happened. Anyway, it is easier to simply change to RNG, or something like that, than to dig that info... I'll change it so we can be sure that it is okay even on Windows 7 and people have piece of mind in this regard.